File: EXAMEN/EXAMEN 2/civitasRuby/lib/main_p4.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'jugador'
require_relative 'jugador_especulador'
require_relative 'diario'
require_relative 'titulo_propiedad'

module Civitas
  class Main_p4
    jugador = Jugador.new("Jugador1")
    propiedad = Titulo_propiedad.new("Moscu", 20 , 1.05, 100 , 200, 125)
    jugador.comprar(propiedad)

    puts jugador.inspect
    puts "Compra la propiedad"
    puts propiedad.inspect
    puts "**********************"

    jugador = Jugador_especulador.new(jugador, 1000)
    puts "Jugador especulador"
    puts jugador.inspect

    puts "Debe ser encarcelado"
    puts  jugador.debe_ser_encarcelado

  end
end

File: EXAMEN/EXAMEN 3/civitasJava/src/civitas/MainP4.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

/**
 *
 * @author mariolopezgonzalez
 */
public class MainP4 {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        Jugador j1 = new Jugador ("jugador1");
        System.out.println("Jugador: " + j1);

        TituloPropiedad titulo = new TituloPropiedad("Moscu", 20 , 1.05f, 100 , 200, 125);
        j1.comprar(titulo);
        System.out.println("Titulo: " + titulo);
        System.out.println("Jugador: " + j1);
        
        System.out.println("**************************");
        
        j1 = new JugadorEspeculador ( j1, 1000);
        System.out.println("Jugador: " + j1);
        System.out.println("Titulo: " + titulo);

        System.out.println("Debe ser encarcelado" + j1.debeSerEncarcelado());
        System.out.println("Jugador: " + j1);
    }
    
}
 